benchmarks/memory_unit_benchmarks.py

```diff
@@ -1,8 +1,15 @@
 import pybamm
+from benchmarks.benchmark_utils import set_random_seed
 import numpy as np
 
 
 class MemCreateExpression:
+    R: pybamm.Parameter
+    model: pybamm.BaseModel
```

Comment:
Out of curiosity, what does this syntax do? Never seen it before

Reply:
It is a way to declare member variables without an init function.
```
variableName: type
```
It also enforces the typing of the variable.

I used it here because the benchmark classes were adding member variables inside other functions, which gives a static analysis warning. It is generally best practice to declare all member variables either in the init() method or in the class like I did here. The main reasoning is to avoid the variables being undefined or assuming that member functions are called in a certain order. Not a major safety concern in these small classes with a very specific purpose, but I still prefer to clean it up if I see it.

Reply:
I should probably add that this syntax is just general typing of python variables:
```python
def some_function(a: str, b: int) -> float:
```
It lets you declare the type of variables with ":" after the variable name and return types with "->" after the function. Typing support is a bit of a pain in Python, but it is useful sometimes.

Reply:
We do need type hints across PyBaMM, maybe a small GSoC project 🙂